hooks/cache_location.py

```diff
@@ -39,9 +42,30 @@ def path_cache(self, project_id, pipeline_configuration_id):
         id will be set to None.
 
         :param project_id: The shotgun id of the project to store caches for
+        :param entry_point: Entry point string to identify the scope for a particular plugin
+                            or integration. For more information,
+                            see :meth:`~sgtk.bootstrap.ToolkitManager.entry_point`. For
+                            non-plugin based toolkit projects, this value is None.
         :param pipeline_configuration_id: The shotgun pipeline config id to store caches for
         :returns: The path to a path cache file. This file should exist when this method returns.
         """
+        # backwards compatibility with custom hooks created before 0.18
+        if hasattr(self, "path_cache") and callable(getattr(self, "path_cache")):
```

Comment:
Trying to wrap my brain around this... This might be available presumably because someone is using new style hooks and this new code is inserted into the inheritance chain? If not using new style hooks but have a custom implementation, what will happen?